admin-dev/themes/new-theme/js/components/contextual-notification.ts

```diff
@@ -0,0 +1,122 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/**
+ * This class is responsible for initiating, setting and getting data related to contextual notifications,
+ * that is to say: should we display the notification related to this key identifier in local storage.
+ * It also displays the notification itself
+ */
+
+class ContextualNotification {
+  // all contextual notification data will be stored under this key in local storage
+  private localStorageKey = 'contextual_notifications';
+
+  constructor() {
+    const notificationList = localStorage.getItem(this.localStorageKey);
+
+    // if the contextual_notifications key doesn't exist in localstorage, we set it as empty
+    if (notificationList === null) {
+      localStorage.setItem(this.localStorageKey, JSON.stringify({}));
+    }
+
+    $(document).on('click', '.contextual-notification .close', (Event) => this.disableNotification(Event));
+  }
+
+  private getNotificationList(): Array<boolean> {
+    const notificationList = localStorage.getItem(this.localStorageKey) ?? '{}';
+
+    return JSON.parse(notificationList);
+  }
+
+  setItem(key: any, value: boolean): void {
+    const notificationList = this.getNotificationList();
+    notificationList[key] = value;
+
+    localStorage.setItem(this.localStorageKey, JSON.stringify(notificationList));
+  }
+
+  getItem(key: any): boolean|null {
+    const notificationList = this.getNotificationList();
+
+    if (key in notificationList) {
+      return notificationList[key];
+    }
+
+    return null;
+  }
+
+  disableNotification(event: any): void {
+    const notificationKey = $(event.target).parent().attr('data-notification-key');
+
+    if (notificationKey !== '') {
+      this.setItem(notificationKey, false);
+    }
+  }
+
+  displayNotification(message: string, key: string): void {
+    const $element = document.createElement('div');
+    $element.classList.add('alert', 'alert-info', 'contextual-notification');
+    $element.setAttribute('data-notification-key', key);
+    $element.innerHTML = `${message}<button type="button" class="close" data-dismiss="alert">&times;</button>`;
+
+    const ajaxConfirmation = document.getElementById('ajax_confirmation');
+
+    if (ajaxConfirmation instanceof HTMLElement && ajaxConfirmation.parentNode instanceof Node) {
+      ajaxConfirmation.parentNode.insertBefore($element, ajaxConfirmation);
+    }
+  }
+}
+
+export default function initContextualNotification(key: string): void {
+  const multistoreHeader = document.querySelector('.header-multishop');
+  const dataAttr = `data-${key}-notification`;
+
+  if (multistoreHeader === null
+    || !(multistoreHeader instanceof HTMLElement)
+    || !multistoreHeader.hasAttribute(dataAttr)) {
+    return;
+  }
+
+  // make localstorage key for this context
+  const contextualNotification = new ContextualNotification();
+
+  const notificationKey = multistoreHeader.dataset.shopId !== undefined
```

Comment:
What about the `All shops` context?

Reply:
No messages must be added in all shops context.

Specs are there : https://github.com/PrestaShop/PrestaShop/issues/23739#issue-838903164

Reply:
OK legitimate, but then maybe it's worth adding a small comment here for future readers
```js
    // Only search notification message for "single shop" or "shop group" context since no notification is needed for "All shops" context
```

Reply:
Oh but actually you should probably check if at least one of the two attributes exist before instanciating the `ContextualNotification` component? Like on Line 99 I guess you can add some conditions:

```js
if (multistoreHeader === null
    || !(multistoreHeader instanceof HTMLElement)
    || !multistoreHeader.hasAttribute(dataAttr)
    || (multistoreHeader.dataset.shopId === undefined && multistoreHeader.dataset.groupId === undefined)) {
    return;
  }
```